### PR TITLE
Object and Filter Support

### PR DESCRIPTION
**Pull request by kooldev**
This PR implements `ReflectionAttribute::getAsObject()` and the `$name` argument of `getAttributes()`. I changed the signature to `getAttributes(?string $name = null, bool $inheritance = false)` to provide both name and instanceof matching. It could make sense to change the bool argument into a flags argument to have a more extensible filter behavior.

This addresses "Required work" items:
- Autoload classes after resolving
- ReflectionAttribute::getAsObject implementation.
- Implement $name filter for attribute or subclass of

I moved all the test cases into a subfolder called `attributes`, this makes it easier to run just these tests using `make test TESTS=./Zend/tests/attributes/` and it is easier to locate them.

### CODE REVIEW

File: ext/reflection/php_reflection.c [hunk 1]
```diff
@@ -6477,41 +6609,160 @@ ZEND_METHOD(reflection_reference, getId)
...
+	}
+
+	if (ce->constructor) {
+		if (FAILURE == call_attribute_constructor(ce, Z_OBJ(obj), args, argc)) {
+			attribute_ctor_cleanup(&obj, args, argc);
+			RETURN_THROWS();
+		}
+	} else if (argc) {
+		attribute_ctor_cleanup(&obj, args, argc);
+		zend_throw_error(NULL, "Annotation class '%s' does not have a constructor, cannot pass arguments", ZSTR_VAL(ce->name));
```
Comment (beberlei) [inline ext/reflection/php_reflection.c]:
```suggestion
		zend_throw_error(NULL, "Attribute class '%s' does not have a constructor, cannot pass arguments", ZSTR_VAL(ce->name));
```

File: ext/reflection/php_reflection.c [hunk 2]
```diff
@@ -3812,15 +3893,32 @@ ZEND_METHOD(reflection_class_constant, getAttributes)
...
+			if (!EG(exception)) {
+				zend_throw_error(NULL, "Class '%s' not found", ZSTR_VAL(name));
+			}
+
+			RETURN_THROWS();
+		}
+
+		name = NULL;
+	}
+
```
Comment (beberlei) [inline ext/reflection/php_reflection.c]: Can you refactor this in a way that its reusable across methods? Its duplicated a bunch of times.

**Review by beberlei — CHANGES_REQUESTED** [review]

File: ext/reflection/php_reflection.c (hunk 1 above)
Comment (kooldev) [reply ext/reflection/php_reflection.c]: Tribute to copy & paste. ;-) Fixed now.

File: ext/reflection/php_reflection.c (hunk 2 above)
Comment (kooldev) [reply ext/reflection/php_reflection.c]: Refactored into a function.

**Review by beberlei — APPROVED** [review]

**Pull request merged**